modules/solid_mechanics/src/neml2/materials/NEML2ParameterDerivativeToMOOSEMaterialProperty.C

```diff
@@ -0,0 +1,98 @@
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NEML2ParameterDerivativeToMOOSEMaterialProperty.h"
+#include "NEML2Utils.h"
+
+registerMooseObject("SolidMechanicsApp", NEML2ParameterDerivativeToRealMOOSEMaterialProperty);
+registerMooseObject("SolidMechanicsApp",
+                    NEML2ParameterDerivativeToStdVectorRealMOOSEMaterialProperty);
+registerMooseObject("SolidMechanicsApp",
+                    NEML2ParameterDerivativeToSymmetricRankTwoTensorMOOSEMaterialProperty);
+registerMooseObject("SolidMechanicsApp",
+                    NEML2ParameterDerivativeToSymmetricRankFourTensorMOOSEMaterialProperty);
+
+#ifndef NEML2_ENABLED
+#define NEML2ParameterDerivativeToMOOSEMaterialPropertyStub(name)                                  \
+  NEML2ObjectStubImplementationOpen(name, Material);                                               \
+  NEML2ObjectStubParam(UserObjectName, "execute_neml2_model_uo");                                  \
+  NEML2ObjectStubParam(MaterialPropertyName, "moose_material_property");                           \
+  NEML2ObjectStubParam(std::string, "neml2_variable");                                             \
+  NEML2ObjectStubParam(std::string, "neml2_parameter_derivative");                                 \
+  NEML2ObjectStubImplementationClose(name, Material)
+
+NEML2ParameterDerivativeToMOOSEMaterialPropertyStub(
+    NEML2ParameterDerivativeToRealMOOSEMaterialProperty);
+NEML2ParameterDerivativeToMOOSEMaterialPropertyStub(
+    NEML2ParameterDerivativeToStdVectorRealMOOSEMaterialProperty);
+NEML2ParameterDerivativeToMOOSEMaterialPropertyStub(
+    NEML2ParameterDerivativeToSymmetricRankTwoTensorMOOSEMaterialProperty);
+NEML2ParameterDerivativeToMOOSEMaterialPropertyStub(
+    NEML2ParameterDerivativeToSymmetricRankFourTensorMOOSEMaterialProperty);
+#else
+
+#include "ExecuteNEML2Model.h"
+
+template <typename T>
+InputParameters
+NEML2ParameterDerivativeToMOOSEMaterialProperty<T>::validParams()
+{
+  auto params = Material::validParams();
+  params.addClassDescription(
+      "This object requests the computation of the derivative of a NEML2 output variable with "
+      "respect to a NEML2 model parameter and saves the derivative as a MOOSE material property");
+
+  params.addRequiredParam<UserObjectName>("execute_neml2_model_uo",
+                                          "User object managing the execution of the NEML2 model.");
+  params.addRequiredParam<MaterialPropertyName>("moose_material_property",
+                                                "MOOSE material property to emit");
+  params.addRequiredParam<std::string>("neml2_variable", "NEML2 output variable to read");
+  params.addRequiredParam<std::string>(
+      "neml2_parameter_derivative", "Return the derivative of neml2_variable with respect to this");
```

Comment:
This should be `neml2_parameter`. with doc string "Parameter to take the derivative with respect to."

Reply:
I take that back!